src/libostree/ostree-repo.c

```diff
@@ -4474,7 +4475,7 @@ ostree_repo_load_file (OstreeRepo         *self,
           if (S_ISLNK (stbuf.st_mode))
             g_file_info_set_symlink_target (*out_file_info, symlink_target);
           else
-            g_assert (S_ISREG (stbuf.st_mode));
+            g_assert (S_ISREG (stbuf.st_mode) || (S_ISCHR(stbuf.st_mode) && stbuf.st_rdev == 0));
```

Comment:
This is the check we want to generalize everywhere I think, as e.g. `_ostree_is_whiteout (&stbuf)`.